cli/src/pcluster/config/cluster_config.py

```diff
@@ -371,17 +372,27 @@ def __init__(self, mount_dir: str, name: str):
     def _register_validators(self, context: ValidatorContext = None):  # noqa: D102 #pylint: disable=unused-argument
         self._register_validator(SharedStorageNameValidator, name=self.name)
 
+    @property
+    def unmanaged(self):
+        """Return true if using existing FSx."""
+        return self.file_system_id is not None
+
     @property
     def file_system_data(self):
         """Return filesystem information if using existing FSx."""
-        if not self.__file_system_data and self.file_system_id:
+        if not self.__file_system_data and self.unmanaged:
             self.__file_system_data = AWSApi.instance().fsx.get_file_systems_info([self.file_system_id])[0]
         return self.__file_system_data
 
+    @property
+    def file_system_subnets(self):
+        """Return list of subnets associated to FSx."""
+        return self.file_system_data.subnet_ids if self.unmanaged else []
+
```

Comment:
This implementation/name might cause confusion in the future: here you are returning the subnet IDs only if the FSx is unmanaged. Is it wise to create this difference?

Reply:
I'd prefer not to ..however `file_system_data` is retrieved only if the FS is unmanaged. This is why I put the check here. 
Moreover I created a property to make it explicit since the previous check wasn't so clear.

I thought about generalizing this also for the managed case, but this was out-of-scope for this PR and also I didn't want to break existing behaviors.